cmd/charm/charmcmd/show.go

```diff
@@ -172,3 +196,125 @@ func handleIncludes(includes []string) (bool, []string, []string) {
+	if ok == false {
+		return nil, errgo.Newf("unexpected type provided: %T", metadata)
+	}
+	var buffer bytes.Buffer
+	sd := newShowData(&buffer, metadata)
+	sd.formatTabular()
+	sd.tw.Flush()
+	return buffer.Bytes(), nil
+}
+
+type showData struct {
+	name            string
+	summary         string
+	owner           string
+	supportedseries []string
+	tags            []string
+	terms           []string
+	promulgated     bool
+	subordinate     bool
+	revision        int
+	bugsUrl         string
+	homePage        string
+	read            []string
+	write           []string
+	channels        []interface{}
+	bundle          bool
+	tw              *tabwriter.Writer
+}
+
+func newShowData(out io.Writer, metadada map[string]interface{}) showData {
+	sd := showData{}
+	sd.tw = tabwriter.NewWriter(out, 0, 8, 8, '\t', 0)
+	sd.revision = int((metadada["id-revision"].(map[string]interface{}))["Revision"].(float64))
+	sd.promulgated = (metadada["promulgated"].(map[string]interface{}))["Promulgated"].(bool)
+	sd.owner = (metadada["owner"].(map[string]interface{}))["User"].(string)
+	sd.bugsUrl = metadada["bugs-url"].(string)
+	sd.homePage = metadada["homepage"].(string)
+	if val, ok := metadada["terms"]; ok {
+		sd.terms = toStringArray(val.([]interface{}))
+	}
+	sd.name = (metadada["id-name"].(map[string]interface{}))["Name"].(string)
+	perms := metadada["perm"].(map[string]interface{})
+	sd.read = toStringArray(perms["Read"].([]interface{}))
+	sd.write = toStringArray(perms["Write"].([]interface{}))
+	sd.channels = (metadada["published"].(map[string]interface{}))["Info"].([]interface{})
+	if val, ok := metadada["charm-metadata"]; ok {
+		charmMetadata := val.(map[string]interface{})
+		sd.summary = charmMetadata["Summary"].(string)
+		sd.supportedseries = toStringArray(charmMetadata["SupportedSeries"].([]interface{}))
+		sd.tags = toStringArray(charmMetadata["Tags"].([]interface{}))
+		sd.subordinate = charmMetadata["Subordinate"].(bool)
+	}
+	if _, ok := metadada["bundle-metadata"]; ok {
+		sd.bundle = true
+	}
+	return sd
+}
+
+func (s *showData) formatTabular() {
+	fmt.Fprintf(s.tw, "%s\t%s", "Name", s.name)
+	fmt.Fprintln(s.tw)
+	fmt.Fprintf(s.tw, "%s\t%s", "Owner", s.owner)
+	fmt.Fprintln(s.tw)
+	fmt.Fprintf(s.tw, "%s\t%d", "Revision", s.revision)
+	fmt.Fprintln(s.tw)
+	s.printCharmMetadata()
+	fmt.Fprintf(s.tw, "%s\t%t", "Promulgated", s.promulgated)
+	fmt.Fprintln(s.tw)
+	fmt.Fprintf(s.tw, "%s\t%s", "Home page", s.homePage)
+	fmt.Fprintln(s.tw)
+	fmt.Fprintf(s.tw, "%s\t%s", "Bugs url", s.bugsUrl)
+	fmt.Fprintln(s.tw)
+	fmt.Fprintf(s.tw, "Read\t%s", strings.Join(s.read, ", "))
+	fmt.Fprintln(s.tw)
+	fmt.Fprintf(s.tw, "Write\t%s", strings.Join(s.write, ", "))
+	fmt.Fprintln(s.tw)
+	if len(s.terms) > 0 {
+		fmt.Fprintf(s.tw, "Terms\t%s", strings.Join(s.terms, ", "))
+		fmt.Fprintln(s.tw)
+	}
+	s.printChannels()
+}
+
+func (s *showData) printChannels() {
+	fmt.Fprintln(s.tw, " \t ")
+	fmt.Fprint(s.tw, "CHANNEL\tCURRENT")
```

Comment:
we would like one more space, if I just do println it will be a new paragrah and not aligned with the first group.